pkg/modules/generators/workload/workload_generator.go

```diff
@@ -405,3 +410,36 @@ func handleFileCreation(c container.Container, uniqueAppName, containerName stri
 	})
 	return
 }
+
+// completeBaseWorkload uses config from workspace to complete the workload base config.
+func completeBaseWorkload(base *workload.Base, config apiv1.GenericConfig) error {
+	replicas, err := workspace.GetIntFromGenericConfig(config, workload.FieldReplicas)
+	if err != nil {
+		return err
+	}
+	if replicas == 0 {
+		replicas = workload.DefaultReplicas
+	}
+	if base.Replicas == 0 {
+		base.Replicas = replicas
+	}
+	labels, err := workspace.GetStringMapFromGenericConfig(config, workload.FieldLabels)
+	if err != nil {
+		return err
+	}
+	if labels != nil {
+		if err = mergo.Merge(&base.Labels, labels); err != nil {
```

Comment:
How does `mergo.Merge` process two maps with the same key? According to our proposal, we are supposed to assure that the labels and annotations from developers won't be overridden by workspace configs.